docs/source/distributed.pipelining.rst

```diff
@@ -4,139 +4,216 @@
 Pipeline Parallelism
 ####################
 
-.. note:: ``torch.distributed.pipelining`` is a package migrated from the `PiPPy <https://github.com/pytorch/PiPPy>`_ project. It is currently in alpha state and under extensive development. For examples that work with our APIs, please refer to PiPPy's `examples <https://github.com/pytorch/PiPPy/tree/main/examples>`_ directory.
+.. note:: ``torch.distributed.pipelining`` is currently in alpha state and under
+development. It is a package migrated from the `PiPPy <https://github.com/pytorch/PiPPy>`_ project.
+
 
 Why Pipeline Parallel?
 **********************
 
-One of the most important techniques for advancing the state of the art in deep learning is scaling. Common techniques for scaling neural networks include *data parallelism*, *tensor/operation parallelism*, and *pipeline parallelism* (or *pipelining*). Pipelining is a technique in which the *code* of the model is partitioned and multiple *micro-batches* execute different parts of the model code concurrently. In many cases, pipeline parallelism can be an effective technique for scaling, in particular for large-scale jobs or bandwidth-limited interconnects. To learn more about pipeline parallelism in deep learning, see `this article <https://www.deepspeed.ai/tutorials/pipeline/>`_.
+One of the most important techniques for advancing the state of the art in deep
+learning is scaling.
+Common techniques for scaling neural networks include *data parallelism*,
+*tensor/operation parallelism*, and *pipeline parallelism*.
+Pipelining is a technique in which the *code* of the model is partitioned and
```

Comment:
i think saying the 'code' is partitioned is not that descriptive.  I would describe it as 'layer-wise' partitioning.

Partitioning of 'code' itself is more of an implementation detail (e.g. with a tracer we split up .forward code sometimes)